src/CabalHelper/Runtime/Main.hs

```diff
@@ -432,12 +456,24 @@ componentsMap lbi _v _distdir f = do
 
 componentOptions' (lbi, v, distdir) inplaceFlag flags rf f = do
   let pd = localPkgDescr lbi
-  componentsMap lbi v distdir $ \c clbi bi -> let
+#if CH_MIN_VERSION_Cabal(2,0,0)
+  includeDirs <- componentsMap lbi v distdir $ \_c clbi bi -> do
+    return (componentUnitId clbi
+           , (componentInternalDeps clbi, hsSourceDirs bi,componentIncludes clbi))
+  let includeDirMap = Map.fromList $ map snd includeDirs
+#endif
+
+  componentsMap lbi v distdir $ \c clbi bi ->
+         let
            outdir = componentOutDir lbi c
            (clbi', adopts) = case flags of
                                _ | not inplaceFlag -> (clbi, mempty)
                                ["--with-inplace"] -> (clbi, mempty)
+#if CH_MIN_VERSION_Cabal(2,0,0)
```

Comment:
I don't like having conditionals in the middle of code like this if not absolutely necessary, can you think of a way to get rid of this one and the one above?

Reply:
Not that I am aware of. Short of putting a dummy `includeDirMap` into the original `removeInplaceDeps`.